### PR TITLE
Handle non-existent files in `lstat`

### LINKED ISSUE #66: fs.lstatSync throws when file doesn't exist.
With the addition of using `fs.lstatSync` 
https://github.com/sindresorhus/trash/commit/126a1dd1532d0bdd236628fb8db505db033d9273, now `trash` errors with
 ```
Error: ENOENT: no such file or directory, lstat './the-file'
    at Object.fs.lstatSync 
```
When a file doesn't exist, where as `fs.existsSync` would return `false`.

Related to #64 *(\cc @jdanford)*

### PR DESCRIPTION
**Pull request by kevva**
Fixes #66.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -16,7 +16,17 @@ module.exports = (iterable, opts) => {
...
-		.filter(fs.lstatSync);
+		.filter(x => {
+			try {
+				return fs.lstatSync(x);
+			} catch (err) {
+				if (err.code === 'ENOENT') {
+					return false;
+				}
+
+				return Promise.reject(err);
```
Comment (sindresorhus) [inline index.js]: You can't return a rejected promise here. Instead, you should throw and rather wrap the module.export method in `Promise.resolve().then()` to catch thrown errors.

Comment (jdanford) [reply index.js]: Also, this predicate is kind of large, what about defining it outside of the filter call?

Comment (kevva) [reply index.js]: > You can't return a rejected promise here.

Hm, that's right, although it does seem to work and reject on failures.

Comment (sindresorhus) [reply index.js]: @kevva Node.js logs the unhandled rejection, sure, but I wouldn't call that "working".

**Pull request merged**